### PR TITLE
SSCS-8596:Invalid page for Caseworker SYA

### PR DESCRIPTION
**Pull request by divisathyan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-8596

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java [hunk 1]
```diff
@@ -95,6 +98,11 @@ public Long submitAppeal(SyaCaseWrapper appeal, String userToken) {
         appeal.setCaseType("draft");
 
         IdamTokens idamTokens = getUserTokens(oauth2Token);
+        if (hasValidCaseworkerRole(idamTokens)) {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java:101]: This same block is repeated multiple times

Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java:101]: I also think 'hasValidCaseworkerRole' is a confusing name. You don't expect an exception to be thrown if something is valid. Something like "isCaseWorker" reads better.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java [hunk 2]
```diff
@@ -91,13 +94,21 @@ public Long submitAppeal(SyaCaseWrapper appeal, String userToken) {
         return caseDetails.getId();
     }
 
+    private void checkHasValidRole(IdamTokens idamTokens) {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java:97]: Minor: by convention this should be moved below with the other private methods

**Review by wmarshid — APPROVED** [review]
just one minor comment - good work!

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java (hunk 2 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/SubmitAppealService.java:97]: Good spot, Done thanks

**Review by gorkem-turan — APPROVED** [review]

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

**Pull request closed without merging**